src/lib/transition_frontier_controller_tests/test_catchup_scheduler.ml

```diff
@@ -0,0 +1,248 @@
+              ~catchup_job_writer ~catchup_breadcrumbs_writer
+          in
+          let randomly_chosen_breadcrumb =
+            Transition_frontier.all_breadcrumbs frontier
+            |> List.permute |> List.hd_exn
+          in
+          let%bind upcoming_breadcrumbs =
+            Deferred.all
+            @@ Quickcheck.random_value
+                 (gen_linear_breadcrumbs ~logger ~size:4
+                    ~accounts_with_secret_keys randomly_chosen_breadcrumb)
+          in
+          let missing_breadcrumb = List.nth_exn upcoming_breadcrumbs 0 in
+          let dangling_breadcrumb1 = List.nth_exn upcoming_breadcrumbs 1 in
+          let dangling_breadcrumb2 = List.nth_exn upcoming_breadcrumbs 2 in
+          let dangling_breadcrumb3 = List.nth_exn upcoming_breadcrumbs 3 in
+          let missing_transition =
+            Transition_frontier.Breadcrumb.transition_with_hash
+              missing_breadcrumb
+          in
+          let dangling_transition1 =
+            Transition_frontier.Breadcrumb.transition_with_hash
+              dangling_breadcrumb1
+          in
+          let dangling_transition2 =
+            Transition_frontier.Breadcrumb.transition_with_hash
+              dangling_breadcrumb2
+          in
+          let dangling_transition3 =
+            Transition_frontier.Breadcrumb.transition_with_hash
+              dangling_breadcrumb3
+          in
+          Catchup_scheduler.watch scheduler ~timeout_duration
+            ~transition:dangling_transition3 ;
+          assert (Catchup_scheduler.has_timeout scheduler dangling_transition3) ;
+          Catchup_scheduler.watch scheduler ~timeout_duration
+            ~transition:dangling_transition2 ;
+          assert (
+            Catchup_scheduler.has_timeout scheduler dangling_transition2
+            && not
+               @@ Catchup_scheduler.has_timeout scheduler dangling_transition3
+          ) ;
+          Catchup_scheduler.watch scheduler ~timeout_duration
+            ~transition:dangling_transition1 ;
+          assert (
+            Catchup_scheduler.has_timeout scheduler dangling_transition1
+            && not
+               @@ Catchup_scheduler.has_timeout scheduler dangling_transition2
+          ) ;
+          Transition_frontier.add_breadcrumb_exn frontier missing_breadcrumb ;
+          Catchup_scheduler.notify scheduler ~transition:missing_transition
+          |> ignore ;
+          assert (Catchup_scheduler.is_empty scheduler) ;
+          let result_ivar = Ivar.create () in
+          Strict_pipe.Reader.iter_without_pushback catchup_breadcrumbs_reader
+            ~f:(fun dangling_rose_trees ->
+              Ivar.fill result_ivar dangling_rose_trees )
+          |> don't_wait_for ;
+          let%map dangling_rose_trees = Ivar.read result_ivar in
+          let received_rose_tree =
+            Rose_tree.T (missing_breadcrumb, dangling_rose_trees)
+          in
+          assert (
```

Comment:
This assert statement is pretty verbose and is used two times. Can you make the assert, a function?